provider/melipayamak/melipayamak.go

```diff
@@ -0,0 +1,67 @@
+package melipayamak
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/messagebird/sachet"
+)
+
+type Config struct {
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
+	From     string `yaml:"from"`
+	Endpoint string `yaml:"endpoint"`
+}
+
+var _ (sachet.Provider) = (*Melipayamak)(nil)
+
+type Melipayamak struct {
+	Config
+	HTTPClient *http.Client
+}
+
+func NewMelipayamak(config Config) *Melipayamak {
+	return &Melipayamak{
+		config,
+		&http.Client{Timeout: time.Second * 20},
+	}
+}
+
+func (mp *Melipayamak) Send(message sachet.Message) error {
+
+	Payload := map[string]string{
+		"username": mp.Username,
+		"password": mp.Password,
+		"to":       strings.Join(message.To, ","),
+		"from":     message.From,
+		"text":     message.Text,
+	}
+	data, _ := json.Marshal(Payload)
+	request, err := http.NewRequest("POST", mp.Endpoint, bytes.NewBuffer(data))
+	if err != nil {
+		return err
+	}
+	request.Header.Set("content-type", "application/json")
+	request.Header.Set("User-Agent", "Sachet")
+	response, err := mp.HTTPClient.Do(request)
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+	body, _ := ioutil.ReadAll(response.Body)
```

Comment:
this is not a big problem but: you only use the body when the status code is not OK. 
So you should ReadAll only inside the if.

Reply:
Yes, you are right. I changed it :+1: